components/aws/sagemaker/common/_utils.py

```diff
@@ -471,11 +471,61 @@ def create_model_request(args):
 
     return request
 
+def endpoint_name_exists(client, endpoint_name):
+  endpoint_exists = False
+  if endpoint_name:
+      endpoints = client.list_endpoints(
+          MaxResults=99,
+          NameContains=endpoint_name,
+      )['Endpoints']
+      for i in range(len(endpoints)):
+          ep = endpoints[i]
+          if (ep['EndpointName'] == endpoint_name):
+            endpoint_exists = True
+            logging.info("Endpoint exists: " + ep['EndpointName'] + " - " + ep['EndpointArn'])
+  return endpoint_exists
+
+def endpoint_config_name_exists(client, endpoint_config_name):
+  ep_config_name_exists = False
+  if endpoint_config_name:
+      if client.describe_endpoint_config(
+          EndpointConfigName=endpoint_config_name,
+          )['EndpointConfigName'] == endpoint_config_name:
+        ep_config_name_exists = True
+        logging.info("Endpoint Config exists: " + endpoint_config_name)
+  return ep_config_name_exists
+
 def deploy_model(client, args):
```

Comment:
Should this be method split into a `deploy_model` and a separate `update_model`? That way the `_utils.py` doesn't handle code flow, just helper methods. (This will make it easier to migrate to the refactored version)

Reply:
Done